src/relay/pass/dispatch_global_func.cc

```diff
@@ -0,0 +1,257 @@
+                                           func->attrs);
+    mod->Add(copied_global_var, new_func);
+    if (pos == conditions[level].size() - 1) {
+      return CallNode::make(copied_global_var, param_exprs);
+    } else {
+      return IfNode::make(conditions[level][pos],
+                          CallNode::make(copied_global_var, param_exprs),
+                          BuildDispatchTree(mod,
+                                            func,
+                                            param_vars,
+                                            param_exprs,
+                                            conditions,
+                                            func_name,
+                                            num_func_copy + 1,
+                                            level,
+                                            pos + 1));
+    }
+  } else {
+    if (pos == conditions[level].size() - 1) {
+      return BuildDispatchTree(mod,
+                               func,
+                               param_vars,
+                               param_exprs,
+                               conditions,
+                               func_name,
+                               num_func_copy,
+                               level + 1,
+                               0);
+    } else {
+      return IfNode::make(conditions[level][pos],
+                          BuildDispatchTree(mod,
+                                            func,
+                                            param_vars,
+                                            param_exprs,
+                                            conditions,
+                                            func_name,
+                                            num_func_copy,
+                                            level + 1,
+                                            0),
+                          BuildDispatchTree(mod,
+                                            func,
+                                            param_vars,
+                                            param_exprs,
+                                            conditions,
+                                            func_name,
+                                            num_func_copy,
+                                            level,
+                                            pos + 1));
+    }
+  }
+}
+
+/*! \brief Dispatch a global var for dynamic input shape
+ *
+ * This function needs the following steps:
+ *
+ *   1. Given symbolic input shape, return a BucketDict with using dispatching
+ *      logic provided by dispatch_func.
+ *
+ *   2. Generate a condition expression list given from BucketDict.
+ *
+ *   3. Recursively build a dispatching tree with condition list.
+ *
+ * Note that currently only continuous range is valid as bucket, and is
+ * represented as a pair of (low, high). If high is -1, it means the high
+ * end is positive infinite.
+ */
+Module DispatchGlobalFunc(const Module& mod,
```

Comment:
Same comment as above, I'm not sure the name makes it clear what is happening when we add a dispatcher.